src/bundle/ApiLoader/SolrEngineFactory.php

```diff
@@ -14,51 +14,22 @@
 use Ibexa\Solr\Gateway\GatewayRegistry;
 use Ibexa\Solr\ResultExtractor;
 
-class SolrEngineFactory
+readonly class SolrEngineFactory
 {
-    private RepositoryConfigurationProviderInterface $repositoryConfigurationProvider;
-
-    /** @var string */
-    private $defaultConnection;
-
-    /** @var string */
-    private $searchEngineClass;
-
-    private GatewayRegistry $gatewayRegistry;
-
-    private CoreFilterRegistry $coreFilterRegistry;
-
-    private Handler $contentHandler;
-
-    private DocumentMapper $documentMapper;
-
-    private ResultExtractor $contentResultExtractor;
-
-    private ResultExtractor $locationResultExtractor;
-
     public function __construct(
-        RepositoryConfigurationProviderInterface $repositoryConfigurationProvider,
-        $defaultConnection,
-        $searchEngineClass,
-        GatewayRegistry $gatewayRegistry,
-        CoreFilterRegistry $coreFilterRegistry,
-        Handler $contentHandler,
-        DocumentMapper $documentMapper,
-        ResultExtractor $contentResultExtractor,
-        ResultExtractor $locationResultExtractor
+        private RepositoryConfigurationProviderInterface $repositoryConfigurationProvider,
+        private string $defaultConnection,
```

Comment:
Looking at the code, there is a possibility that this value might be `null`. That, or there is an unnecessary check in `buildEngine()` method.